src/tests/nativeaot/SmokeTests/DotnetJs/wwwroot/main.js

```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { dotnet, exit } from './dotnet.js'
+
+const { runMain } = await dotnet
+    .withApplicationArguments("A", "B", "C")
+    .create();
+
+var result = await runMain();
+console.log(`Exit code ${result}`);
+exit(result == 42 ? 100 : 1);
```

Comment:
It would be more direct to return the exist from from the test (with appropriate changes to the test).
```suggestion
var result = await runMain();
console.log(`Exit code ${result}`);
exit(result);
```